Horizontal space at value relation widget at multiple selection

signedav opened the pull request:
## Description
When in the value relation widget allow multiple selection is active, the widget was displayed as QListWidget where you could select the related items. This could have been not very nice when there were a lot of items. It's now a QTableWidged with multiple columns. It's much more comfortable to handle a lot of items and in case there are very few items it can be made very small and we can save some space. The number of columns is configurable.

### Now
![multiselect_is](https://user-images.githubusercontent.com/28384354/34377559-e4eff2c2-eaf1-11e7-86a8-ab33d7fea8d2.gif)
(if only one column it looks the same like before)

### Before
![multiselect_was](https://user-images.githubusercontent.com/28384354/34377491-8562ace6-eaf1-11e7-8d4b-ed76ca51c43a.png)

pblottiere committed:
Fixes #17576 Avoid crash during node editor cleanup

pblottiere committed:
Fixes #17574 Add Z support in AddCircularString map tools

pblottiere committed:
Add Z support for AddCircle map tools

pblottiere committed:
Fixes crash in map tool

pblottiere committed:
Rename pointFromPointXY to fromPointXY

pblottiere committed:
Add Z support for AddEllipse map tools

pblottiere committed:
Add Z support for AddRectangle map tools

pblottiere committed:
Add Z support to AddRegularPolygon map tools

3nids committed:
[sipify] also create links in docstrings

3nids committed:
do not create link for current class

3nids committed:
run sipify

strk committed:
Allow age expression to represent time intervals above 130 years (#5914)

Uses a 64bit integer instead of 32bit one for seconds,
meaning about 584 billions of years rather than 136...

Closes #17681

Includes test

3nids committed:
[doxygen] add test to ensure code snippet use \code command

3nids committed:
[sipify] habndle python code snippets

3nids committed:
run sipify for code snippets

3nids committed:
fix code snippet and do not display cpp code in sip files

3nids committed:
add cpp to code type

3nids committed:
fix code descriptor

3nids committed:
fix empty lines with spaces

nyalldawson committed:
Add transform context to processing context

nyalldawson committed:
Require a transform context when setting the destination crs for
a feature request

Forces correct datum handling for these requests, and removes
a few more uses of the deprecated/datum unaware transforms.

nyalldawson committed:
Fix missing transform contexts for QgsDistanceArea

QgsDistanceArea.setSourceCrs() now requires a QgsTransformContext
argument.

nirvn committed:
Use qgsfilewidget for the delimited text prodivder source select (#5920)

Gustry committed:
remove CRS picker and allow only layer CRS or provider CRS in metadata editor (#5915)

Gustry committed:
Crs (#5908)

* fix display of CRS if invalid CRS

* fix #17455 set an empty CRS if the read CRS from XML is wrong

* add warning about empty CRS node when reading XML

borysiasty committed:
Fix group id of GDAL algorithms. Follows up 53bc147dca00

m-kuhn committed:
Followup 9a5435f: Fix sip files

signedav committed:
In the GUI it's mNofColumns defining the number of columns - only available on "AllowMulti"

The ValueRelationWidget is a QTableWidget instead of a combobox. The QListWidged is still implemented, but obsolete. It's like a QTableWidget with one column.

1 Remove the QListWidget stuff in the ValueRelationWidget
2 Finetuning on functions and GUI
3 Check out what has to be done in QgsValueRelationSearchWidgetWrapper
4 Selecting feature by enter first letter only works for the current column and not the entire table (has nothing to do with point 3).

signedav committed:
In the GUI it's mNofColumns defining the number of columns - only available on "AllowMulti"

The ValueRelationWidget is a QTableWidget instead of a combobox. The QListWidged is still implemented, but obsolete. It's like a QTableWidget with one column.

1 Remove the QListWidget stuff in the ValueRelationWidget
2 Finetuning on functions and GUI
3 Check out what has to be done in QgsValueRelationSearchWidgetWrapper
4 Selecting feature by enter first letter only works for the current column and not the entire table (has nothing to do with point 3).

signedav committed:
In the GUI it's mNofColumns defining the number of columns - only available on "AllowMulti"

The ValueRelationWidget is a QTableWidget instead of a combobox. The QListWidged is still implemented, but obsolete. It's like a QTableWidget with one column.

1 Remove the QListWidget stuff in the ValueRelationWidget
2 Finetuning on functions and GUI
3 Check out what has to be done in QgsValueRelationSearchWidgetWrapper
4 Selecting feature by enter first letter only works for the current column and not the entire table (has nothing to do with point 3).

signedav committed:
Replaced completely QListWidget with QTableWidget

Clean code. In qgsvaluerelationsearchwidgetwrapper was old code never used with the QListWidged.
Tests are now for QTableWidget instead of QListWidget.

signedav committed:
Replaced completely QListWidget with QTableWidget

Clean code. In qgsvaluerelationsearchwidgetwrapper was old code never used with the QListWidged.
Tests are now for QTableWidget instead of QListWidget.

signedav committed:
Replaced completely QListWidget with QTableWidget

Clean code. In qgsvaluerelationsearchwidgetwrapper was old code never used with the QListWidged.
Tests are now for QTableWidget instead of QListWidget.

m-kuhn commented on src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.cpp:
```diff
@@ -112,17 +117,33 @@ void QgsValueRelationWidgetWrapper::initWidget( QWidget *editor )
+    mTableWidget->setColumnCount( config( QStringLiteral( "NofColumns" ) ).toInt() );
+
+    int row = 0, column = 0;
+    for ( const QgsValueRelationFieldFormatter::ValueRelationItem &element : mCache )
```
needs a const copy of mCache

m-kuhn commented on src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.cpp:
```diff
@@ -112,17 +117,33 @@ void QgsValueRelationWidgetWrapper::initWidget( QWidget *editor )
+    mTableWidget->setShowGrid( false );
+    mTableWidget->setEditTriggers( QAbstractItemView::NoEditTriggers );
+    mTableWidget->setSelectionMode( QAbstractItemView::NoSelection );
+    mTableWidget->setRowCount( ( mCache.size() + config( QStringLiteral( "NofColumns" ) ).toInt() - 1 ) / config( QStringLiteral( "NofColumns" ) ).toInt() );
```
Can you call that "ColumnCount"?

m-kuhn commented:
@signedav can you rebase and push, https://github.com/qgis/QGIS/pull/5853/commits/90acb5690e69f890ac7f79a00b50b8c380fc928d could potentially also have solved this issue

signedav committed:
tests for tablewidget - fixed setEnabled for test case

tests need to be done for tableWidget instead of listWidget
setEnabled has to go for the count of columns instead the configured column number - otherwise tests fail

signedav committed:
tests for tablewidget - fixed setEnabled for test case

tests need to be done for tableWidget instead of listWidget
setEnabled has to go for the count of columns instead the configured column number - otherwise tests fail

signedav committed:
tests for tablewidget - fixed setEnabled for test case

tests need to be done for tableWidget instead of listWidget
setEnabled has to go for the count of columns instead the configured column number - otherwise tests fail

pblottiere committed:
Add test for z values on QgsMapToolAddFeature with LineStringZ

pblottiere committed:
Fixes variables' names

pblottiere committed:
Add some tests for circular string map tools

pblottiere committed:
Add some tests for circle map tools

pblottiere committed:
Add some tests for ellipse map tools

pblottiere committed:
Add some tests for ellipse map tools

pblottiere committed:
Add some tests for regular polygon map tools

pblottiere committed:
The first point in 3D is used to propagate Z dimension

pblottiere committed:
Some clean

pblottiere committed:
Take into account z dimension from center points

pblottiere committed:
Update documentation

pblottiere committed:
Drop z dimension in case of circle from tangential points

pblottiere committed:
Fix spelling check and sip binding

DelazJ committed:
[DB Manager]Allow preset query combobox expansion

Fixes #17933
Also add some window title capitalisation

DelazJ committed:
Better widget's alignment

DelazJ committed:
Update some button labels

DelazJ committed:
Enable rule-based dialog instead of widget all time

DelazJ committed:
Add help button and connect to user manual

DelazJ committed:
Add scrollbar to only dialog-display mode

and not in widget mode
Sur la branche rulebaseDialog
	modifié :         src/app/nodetool/qgsnodetool.h
aucune modification n'a été ajoutée à la validation (utilisez "git add" ou

DelazJ committed:
Set tab order

pblottiere committed:
Fixes build

DelazJ committed:
Merge branch 'master' into rulebaseDialog

mj10777 committed:
Added QgsLayerMetadata logic to QgsDataProvider. Added setMetadata in QgsVector/RasterLayer::setDataProvider. Corrections in QgsMetadataWidget with added get/setMetadata and hasChanged functions plus checking for set mLayer pointer before use.

nirvn committed:
fix disabled map unit  widgets when opening project prop. dialog
(fixes #17208)

manisandro committed:
Don't set focusProxy in QgsCollapsibleGroupBoxBasic when compiled against Qt5.10+, it causes crashes

NathanW2 committed:
Lets try fix this

NathanW2 committed:
Move settings to resource folder

NathanW2 committed:
FIx

NathanW2 committed:
Don't call init in QgsApplication

nyalldawson committed:
Make QgsLocator more thread safe

- add a clone() method to filters, and always search using the
clone instead of the original filter
- add a prepare() method to filters, which is always run in the
main thread and can be used to prepare the filter for safe
background execution (e.g. creating feature iterators in advance)
- don't use QtConcurrent to perform searches in background threads,
since it is not safe to use with QObjects
- instead manually create threads and ensure that cloned objects
are always moved to the thread that they will run in, to ensure
that they correctly have thread affinity with the thread in which
they are executed

elpaso committed:
Merge branch 'settings_17670' of https://github.com/NathanW2/QGIS into NathanW2-settings_17670

nyalldawson committed:
Add missing docs

nyalldawson committed:
Fix error when running gdal algorithms through locator

nyalldawson committed:
Allow certain locator filters to operate on the main thread

Some filters are fast enough to return results that it's overkill
to run them in a background thread - add a flag to these filters
to allow them to run (blocking) in the main thread instead.

nyalldawson committed:
Simplify code

nyalldawson committed:
Doxygen

DelazJ committed:
Fix message boxes titles

DelazJ committed:
Revert arg simplification attempt

nyalldawson committed:
[dbmanager] Workaround Python error when vacuuming spatialite table

Workaround https://bugs.python.org/issue28518

Fixes #18079

nyalldawson committed:
[afs] Handle layers where the id field is not named "objectid"

nyalldawson committed:
[afs] Don't list layers with no Query capability in AFS source select

Since these layers cannot be queried or added to the map and should
be added as mapserver layers instead.

nyalldawson committed:
[processing] Don't replace " with ' when handling layer paths

Since netcdf, and possibly other gdal drivers, use layer uris
of the format NETCDF:"/tmp/test.nc":var1 we can't safely
remove or reformat these quotations.

elpaso committed:
Spelling and docs

elpaso committed:
Add init() to QGSTEST_MAIN

Now that it's not called from ctor anymore

elpaso committed:
Remove debug output from stdout

elpaso committed:
[server] Stop messing with app initialization paths

elpaso committed:
[bugfix] Restore color picker from layer styling panel

Fixes #17721

Successfully tested on Windows 10 and Linux with Qt 5.10

signedav committed:
Merge branch 'work_multiselect' of github.com:signedav/QGIS into work_multiselect

signedav committed:
Merge branch 'master' into work_multiselect

signedav committed:
fix wrong merging of listwidget instead of tablewidget

jef-n committed:
translation string fixes and cleanups

nyalldawson committed:
[afs] Fix provider incorrectly overrides provider name and description
with layer name and description

jef-n committed:
fix 6dfedc38

nyalldawson committed:
Merge pull request #6353 from nyalldawson/proc_normalize

[processing] Don't replace " with ' when handling layer paths

nyalldawson committed:
Fix some undersized widgets on hidpi win builds

nyalldawson committed:
Merge pull request #6316 from nyalldawson/locator_thread

Make QgsLocator more thread safe

nirvn committed:
followup 6dfedc3, fix processing tests

nyalldawson committed:
[processing] Add easy method to retrieve layers from context

Allows python algorithms to call

layer = context.getMapLayer(other_alg_results['OUTPUT'] )

jef-n committed:
debian nightlies: ignore dash submission errors

elpaso committed:
Check for application instance before calling init

elpaso committed:
Merge pull request #6354 from elpaso/NathanW2-settings_17670

Nathan w2 settings 17670

elpaso committed:
Change button color when picking around

So that we have a consistent behavior with the
Sample Color action.

nyalldawson committed:
[processing] Always report errors if features cannot be written
to a destination

Before we silently ignored these - now algorithms will automatically
push errors to the log if a feature cannot be written to a sink (no
changes to algorithms or special handling required)

nyalldawson committed:
Add provider test to ensure that features added with missing attributes
are transparently padded out with NULL attributes to the required fields
length

Currently the behavior is inconsistent - some providers reject these
features, others pad them out, and worse -- some add them with
missing attributes (memory provider), causing ALL sorts of flow-on,
difficult to debug issues.

pblottiere committed:
Z value from snapped point is kept

elpaso committed:
Merge pull request #6356 from elpaso/bugfix-17721-color-picker

[bugfix] Restore color picker from layer styling panel

nyalldawson committed:
Fix memory provider handling of missing attributes when adding features

Ensure that features added to a memory provider (and returned when
fetching features from a memory provider) always have the correct
number of attributes present.

Fixes many random behavior bugs when working with memory providers.

nyalldawson committed:
Update test

elpaso committed:
Fix raster layer names in geopackage browser data items

3nids committed:
[offset tool] handle rings and allow selecting by area

* correctly handles rings and parts
* allow to select polygon by area (not only edge)

nyalldawson committed:
Add provider test to ensure that adding features to a provider
with MORE attributes than expected results in a failure

We need to flag these and not silently discard the extra
attributes resulting in loss of information -- if thisi
situation occurs there's a deeper bug present whic
 needs to be addressed.

nyalldawson committed:
Ensure ogr, postgis and memory providers reject features
with too many attributes

nyalldawson committed:
Add provider test to ensure providers reject adding features
with geometry type mismatch

nyalldawson committed:
Fix adding features with geometry to geometryless memory layer

3nids committed:
[settings] add method to get value for a setting associated to an enum

this will make sure the returned value is actually an existing entry of the enum

3nids committed:
make QgsGeometry a Q_GADGET

so Q_ENUM can be used

nyalldawson committed:
Fix test

nyalldawson committed:
Rollback rejection of features with too many attributes

Turns out editable joins rely on this situation. Instead change
the providers to warn on this occurance, and make the memory
provider alone truncate the extra attributes (since it doesn't
have an external backend or disk based format which natively
applies this truncation)

nyalldawson committed:
Add test that providers always accept features with null geometry

nyalldawson committed:
Correct test

nyalldawson committed:
Start implementing the provider test suite for ArcGIS Feature Server provider

nyalldawson committed:
[afs] Correctly return an invalid feature from closed iterators

jef-n committed:
Adapt packaging to c988dd291

nyalldawson committed:
Try to fix postgres provider handling of missing attributes

DelazJ committed:
More message box fixes

Commit:
Updated and vectorized several icons (#6362)

mActionFileExit
repositoryUnavailable
repositoryDisabled
repositoryConnected
mActionRefresh
mIconDelete
mActionRemoveSelectedFeature
mActionFromSelectedFeature

jef-n committed:
Merge pull request #6329 from DelazJ/QMessageBoxTitles

Fix some message boxes titles

jef-n committed:
more translation string fixes and updates

jef-n committed:
string truncations now uses ellipses (followup c3573b163)

3nids committed:
add test for QgsSettings::enumSettingValue

DelazJ committed:
Fix message box title

3nids committed:
run sip_include.sh

jef-n committed:
drastically speedup update_ts.sh with Qt5's lupdate

3nids committed:
Merge pull request #6364 from 3nids/setting_enum_value

[settings] add method to get value for a setting associated to an enum

slarosa committed:
[processing] fix variable name

3nids committed:
Merge pull request #6365 from 3nids/geom_gadget

make QgsGeometry a Q_GADGET

3nids committed:
make QgsTolerance a Q_GADGET

to use Q_ENUM

3nids committed:
improve test for QgsSettings::enumSettingValue

3nids committed:
test geometry validity before applying offset

3nids committed:
[offset tool] fix #18115 improved offset calculation

* use QgsGeometry::contains for polygons (contains => negative offset)
* use the geometry part and not the whole geometry to find closest segment

3nids committed:
fix spelling

slarosa committed:
[welcomepage] fixes #18118 remove last item causes a crash

nyalldawson committed:
Exempt AFS provider from certain provider tests

nyalldawson committed:
Simplify QgsVectorDataProvider::fillMinMaxCache()

nyalldawson committed:
Don't try to run provider tests which require subsetStrings on providers which don't support them

nyalldawson committed:
Avoid some detaches in AFS iterator

nyalldawson committed:
[afs] Correctly convert attribute values to field type

Avoids AFS provider returning double values for integer
and long field types

nyalldawson committed:
Better geometry memory management for afs provider

nyalldawson committed:
[afs] Remove unused code

nyalldawson committed:
[afs] Fix provider ignores FilterFids requests and returns
incorrect features

nyalldawson committed:
[afs] Fix crash when requesting invalid feature ID

nyalldawson committed:
Make feature source/provider test handle additional fields

nyalldawson committed:
[afs] Ensure that null attributes are correctly mapped for PyQGIS

nyalldawson committed:
More provider check handling of extra fields

nirvn committed:
[ui] update mkuidefaults.py script

nirvn committed:
[ui] update default windows state and geometry

nirvn committed:
[layout] use nicer default state and geometry

pblottiere committed:
Merge pull request #5823 from pblottiere/bugfix_curvedstrings_z

[bugfix] Fixes #17576 and #17574 Add Z support and fixes crashes

nirvn committed:
[needs-docs][layout] rename command history to undo history

jef-n committed:
INSTALL is a output file (followup 1dd6db384f)

jef-n committed:
Merge pull request #6373 from DelazJ/QMessageBoxTitles

Fix message box title

jef-n committed:
update translation strings

jef-n committed:
update_ts.sh: fix selective download of translations

3nids committed:
revert adding tolerance to QgsSnappingUtils::snapToCurrentLayer

since it already uses digitizing settings

3nids committed:
snapToCurrentLayer correctly handles Area type

jef-n committed:
add missing copyright headers

3nids committed:
add test for QgsPointLocator::nearestArea

nyalldawson committed:
Fix accidental test change

nyalldawson committed:
[afs] Don't lock shared data whilst fetching remote data

Since the fetch can be slow, there's no need to lock other
afs threads while we wait for a particular set of results to
be fetched.

nyalldawson committed:
Try to fix test which fails only on Travis

nyalldawson committed:
Remove unused QgsVectorLayerInterruptionCheckerDuringCountSymbolFeatures class

nyalldawson committed:
Remove QgsInterruptionChecker and replace with QgsFeedback use

There's little need for two classes with similar goals here.

3nids committed:
tweak match from nearestEdge to return an Area type

also make that VisitorArea return the given point as the match point

3nids committed:
fix double/int comparison

nyalldawson committed:
[afs] Fix/optimise handling of filter rect feature requests

Before a filter rect request would usually force fetching every
single feature from the server before the request could be
complete.

Instead, if a filter rect is passed we first obtain a list
from the server of matching features within this rect, and
then iterate only over those.

Fixes broken (multi-minute hang) identify tool use on AFS
layers.

nyalldawson committed:
[afs] Add unit test for non OBJECTID named id fields

nyalldawson committed:
Prepare expression for faster field calculation in attribute table dialog

NathanW2 committed:
[FIX] Fix loading projects with . in name

At least on Windows 2.18.qgs is a valid name for a file.

nyalldawson committed:
Fix uuid() function returns the same uuid

Fixes #18130

nyalldawson committed:
[layouts] Remove use of QStyleOptionGraphicsItem from QgsLayoutItem API

Turns out the useful methods in QStyleOptionGraphicsItem were deprecated
in Qt 5.10. To future protect our API this replaces all public
use of QStyleOptionGraphicsItem, so that we're free to revise
the internals of how we use QStyleOptionGraphicsItem without breaking
the public QGIS API.

nyalldawson committed:
[layouts] Move responsibility for matching map colors to project
background color to app

This fits better in app - the silent auto setting of background
colors should not apply to scripts manually creating QgsLayoutItemMap
items.

nyalldawson committed:
Fix geometry checker cannot export errors to gpkg

Fixes #18128

Also exposes other format choices in dialog

NathanW2 committed:
Better sort order for browser node

nyalldawson committed:
Use first feature in layer to generate symbol preview icon in widget

Uses the first feature found in a layer in order to create a better
preview for the symbol in the symbol selector dialog. This allows
data defined settings to be evaluated correctly (at least, for the
first feature) and avoids missing previews due to missing
attribute values when data defined settings are present.

Fixes #17061

nyalldawson committed:
Merge pull request #6375 from nyalldawson/interupt

Remove QgsInterruptionChecker and replace with QgsFeedback

elpaso committed:
Merge pull request #6167 from DelazJ/DBManagerQueryDialog

[DB Manager]Allow preset query combobox to expand

jef-n committed:
fix e606b795d

nirvn committed:
[layouts] improve right-click popup menu
- as with the main window, show bold titles
- sort toolbars and panels alphabetically

elpaso committed:
Merge pull request #6187 from DelazJ/rulebaseDialog

Show "Edit rule" dialog instead of widget when in Layer properties dialog

nirvn committed:
[layouts] sort panels and toolbars sub-menus

nirvn committed:
[layouts] add missing items in the toolbars sub-menu

nirvn committed:
[layouts] hide atlas/report items in the window right-click menu

nyalldawson committed:
[afs] Support interruption requests on feature iterator

Results in a huge increase in responsiveness when working with
large AFS layers

nirvn committed:
[layouts] improve default window state for reports

nyalldawson committed:
[afs] Correctly advertise SelectAtId capability

Big speedup for attribute table loading with provider

NathanW2 committed:
Change order of XYZ tiles

NathanW2 committed:
Merge pull request #6379 from NathanW2/browser_sort_order

Better sort order for browser node

3nids committed:
[sipify] fix Q_FLAG declaration

3nids committed:
extend QgsSettings::enumSettingValue to handle flags (with test)

3nids committed:
use Qgsettins::enumSettingValue where possible

3nids committed:
Qt 5.5 prefers Q_FLAG over Q_FLAGS

this also allow to use QMetaEnum for flags

3nids committed:
[offset] meter limit min to 1, quadrant segments min to 1

manisandro committed:
Merge pull request #6301 from manisandro/qt5.10

Initial workaround for Qt5.10 instability

3nids committed:
[offset] warn user when layer has Z/M geometries

jef-n committed:
api doc: add link to python version

jef-n committed:
api doc: 2.18 is also LTR

3nids committed:
[delete part] use snapping index to find part

jef-n committed:
fix some references to PROJ.4

jef-n committed:
fix build with grass

jef-n committed:
proj5: show proj version qgis was built with and the version it's running with

nyalldawson committed:
[processing] Select by expression/attribute can run on geometryless layers

nyalldawson committed:
[processing] Fix raster layers shown as inputs for certain vector algs

Fixes #18148, #18149

nyalldawson committed:
[processing] Fix algorithm help link in modeler

Fixes #18147

nyalldawson committed:
Merge pull request #6369 from slarosa/fix_18118_crash

[welcomepage] fixes #18118 remove last item causes a crash

nyalldawson committed:
Followup 0548f4, fixes to layer metadata and metadata widget

Address outstanding review requests so that this PR can be merged

nyalldawson committed:
[afs] Don't request intersecting features from server if all
features are already locally cached

nyalldawson committed:
Fix build

nyalldawson committed:
Avoid redundant error

nyalldawson committed:
[afs] More responsive cancelation for feature requests

nyalldawson committed:
Add a bounding box intersection test to QgsGeometry

We only previously had methods for exact intersections - this
commit adds a new QgsGeometry.boundingBoxIntersects() method
which can be used to test if just the bounding boxes of
geometries/rectangles intersect.

It's fast, and doesn't care about invalid geometries (unlike
the exact intersects checks)

nyalldawson committed:
[afs] Use faster bounding box checks where possible

nyalldawson committed:
Merge pull request #6390 from nyalldawson/provider_metadata

Support reading layer metadata from data provider

nyalldawson committed:
Fix links are doubled in metadata widget

nyalldawson committed:
[afs] Auto populate layer metadata from provider metadata

nyalldawson committed:
[afs] Add unit test for metadata

nyalldawson committed:
Spelling

nyalldawson committed:
Merge pull request #6370 from nyalldawson/afs

ArcGIS feature server: implement provider test suite and fix issues

nyalldawson committed:
Fix memory leak in task manager widget

3nids committed:
Revert "fix build with grass"

This reverts commit e5c9092a4cc5f4ddc0e424294a6e8b3fc44ed589.

3nids committed:
add Q_ENUM for QgsRasterProjector::Precision

3nids committed:
Qt 5.7 dropped pyqtWrapperType (#6393)

version would need double check though

rldhont committed:
[BUGFIX][Server] WFS GetFeature GML: segfault with geometry null

signedav committed:
Merge branch 'work_multiselect' of github.com:signedav/QGIS into work_multiselect
